CONTRIBUTING.md

```diff
@@ -22,9 +22,28 @@ Our coding conventions follow two alternative patterns for synchronization withi
 #### Mutex best practices
 
 * Use a `RWMutex` and separate between read locks and write locks. Do not assume that reading without a lock is safe.
-* All state variables that are protected by the mutex should have a name suffix `UnderMutex` and be declared immediately after the mutex.
-* Unlocks of the mutex should be done using `defer mutex.Unlock()` and appear immediately after the locks. Function scope should be designed according to this principle to avoid locking the mutex for too long.
-* Locks should be for as short as possible (only while the data is accessed). Never make a service RPC call or an IO call when the mutex is locked.
+* All state variables that are protected by the a mutex should be grouped in an anonymous struct that embeds the mutex.
+* Unlocks of the mutex should be done using `defer mutex.Unlock()` and appear immediately after the locks. Function scope should be designed according to this principle to avoid locking the mutex for too long. It's best to create mutex access methods that lock the mutex, defer the unlock, perform the read/write and return.
+* Locks should be for as short as possible (only while the data is accessed). Never make an out-of-service-bound call or an IO call when the mutex is locked.
+
+```golang
```

Comment:
I shortened the code snippet to contain only relevant lines, and moved the "Beware" section below it because it is not related to the code.